test/unit/org/apache/cassandra/cql3/validation/operations/TruncateTest.java

```diff
@@ -45,4 +59,15 @@ public void testTruncate() throws Throwable
             assertEmpty(execute("SELECT * FROM %s"));
         }
     }
+
+    @Test
+    public void testRemoteTruncateStmt() throws Throwable
+    {
+        emptyTruncate = true;
```

Comment:
note that while having a provider which returns a value of a static mutable field, you can fill that field with a mock, and then verify whether the mock was really used, something more or less like:

```java
EmptyTruncateStatementProvider.provider = Mockito.mock(TruncateStatement.class);
...
execute("TRUNCATE TABLE %s");
Mockito.verify(EmptyTruncateStatementProvider.provider).executeLocally...
...
```

Also, this way you need only the provider for testing, `EmptyTruncateStatement` class is not needed at all because it is enough to have a mock of it - wdyt?

Reply:
The aim of the test is to check ability to provide another implementation of `TruncateStatement`, thus I see it is correct to provide it together with the provider.